### PR TITLE
Fix stake selection for the Data Plane

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem
Avalanche isn't using the correct "stakes", instead its just using a node's bank balance.

#### Summary of Changes
Updated data-plane's broadcast and retransmit to use stakes from a Bank.

Depends-on: #2860

### CODE REVIEW

File: src/broadcast_service.rs [hunk 1]
```diff
@@ -227,7 +227,10 @@ impl BroadcastService {
             if exit_signal.load(Ordering::Relaxed) {
                 return BroadcastServiceReturnType::ExitSignal;
             }
-            let mut broadcast_table = cluster_info.read().unwrap().sorted_tvu_peers(&bank);
+            let mut broadcast_table = cluster_info
+                .read()
+                .unwrap()
+                .sorted_tvu_peers(&bank.get_stakes());
```
Comment (garious) [inline src/broadcast_service.rs]: This function, `get_stakes()` is misleading. How about renaming that to `staked_validators()` and documenting it with something like: a staker with non-zero balance has delegated validation do the `node_id` returned in the `(node_id, stake)` pair.

File: tests/cluster_info.rs [hunk 2]
```diff
@@ -37,19 +35,13 @@ fn run_simulation(num_nodes: u64, fanout: usize, hood_size: usize) {
     // set timeout to 5 minutes
     let timeout = 60 * 5;
 
-    // math yo
```
Comment (garious) [inline tests/cluster_info.rs]: less fun please

File: src/broadcast_service.rs (hunk 1 above)
Comment (sagar-solana) [reply src/broadcast_service.rs]: @garious, Yeah I was thinking about this on the drive home. I think it should just be "staked_nodes".
Can you help me understand the second part of your comment? A staker with no balance has delegated validation?

Comment (garious) [reply src/broadcast_service.rs]: Not "no balance". "non-zero balance".  Technically the Vote program can be referencing a staker that has emptied its account. We need to make sure to drop those.

**Pull request merged**